src/savegame_json.cpp

```diff
@@ -859,6 +859,18 @@ void npc::load(JsonObject &data)
         attitude = npc_attitude(atttmp);
     }
 
+    if ( data.read( "companion_mission", comp_miss) ) {
```

Comment:
So I converted the three values that I added to just:

    data.read( "companion_mission", companion_mission );
    data.read( "companion_mission_time", comp_miss_t);
    data.read( "restock", stock) );

And all of my companions automatically advanced in the number of hours they were on mission between saves... I'm just going to leave it be since I'm not familiar enough with the json saving and doing something unnecessary is better than introducing more bugs.